################################################################################
SHORT DESCRIPTION:
################################################################################
Allow stubdomains to have multiple disks. Derive the correct device path.

################################################################################
LONG DESCRIPTION:
################################################################################

################################################################################
CHANGELOG
################################################################################
Derived from original patch by:
Jed Lejosne <lejosnej@ainfosec.com>

################################################################################
REMOVAL
################################################################################

################################################################################
UPSTREAM PLAN
################################################################################
Yes with upstreaming Linux stubdomain support.

################################################################################
INTERNAL DEPENDENCIES
################################################################################
stubdomain RFC patches

################################################################################
PATCHES
################################################################################
--- a/tools/libxl/libxl_dm.c
+++ b/tools/libxl/libxl_dm.c
@@ -1580,7 +1580,10 @@ static int libxl__build_device_model_arg
                         return ERROR_INVAL;
                     }
                     if (b_info->stubdomain_version == LIBXL_STUBDOMAIN_VERSION_LINUX) {
-                        target_path = "/dev/xvda";
+                        target_path = (char *[]) {"/dev/xvda",
+                                                  "/dev/xvdb",
+                                                  "/dev/xvdc",
+                                                  "/dev/xvdd"} [disk];
                         format = qemu_disk_format_string(LIBXL_DISK_FORMAT_HOST_DEVICE);
                     }
                     if (colo_mode == LIBXL__COLO_SECONDARY) {
